File: apps/android/lib/models/attendance.dart
class LocalAttendanceRecord {
  final String id;
  final String studentId;
  final String studentName;
  final String sessionId;
  final String status; // present, absent, late, excused
  final bool isSynced;
  final String timestamp;

  LocalAttendanceRecord({
    required this.id,
    required this.studentId,
    required this.studentName,
    required this.sessionId,
    required this.status,
    this.isSynced = false,
    required this.timestamp,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'student_id': studentId,
      'student_name': studentName,
      'session_id': sessionId,
      'status': status,
      'is_synced': isSynced ? 1 : 0,
      'timestamp': timestamp,
    };
  }

  factory LocalAttendanceRecord.fromMap(Map<String, dynamic> map) {
    return LocalAttendanceRecord(
      id: map['id'],
      studentId: map['student_id'],
      studentName: map['student_name'],
      sessionId: map['session_id'],
      status: map['status'],
      isSynced: map['is_synced'] == 1,
      timestamp: map['timestamp'],
    );
  }
}

File: server/app/schemas/attendance.py
# JMO Management System — Attendance Schemas
from pydantic import BaseModel, Field
from typing import Optional, List
from uuid import UUID
from app.schemas.common import PaginatedResponse
from app.models.attendance import AttendanceStatus


class AttendanceRecord(BaseModel):
    id: UUID
    student_id: UUID
    session_id: UUID
    status: str
    recorded_by: UUID
    sync_source: Optional[str]
    recorded_at: str

    class Config:
        from_attributes = True


class BulkAttendanceRequest(BaseModel):
    session_id: UUID
    records: List[AttendanceRecord]


class AttendanceSyncRecord(BaseModel):
    session_id: UUID
    student_id: UUID
    status: str = Field(..., pattern="^(present|absent|late|excused)$")
    recorded_at: str
    client_id: UUID


class AttendanceSyncRequest(BaseModel):
    records: List[AttendanceSyncRecord]


class AttendanceSyncResponse(BaseModel):
    synced: int
    conflicts: int